### PR TITLE
ToggleTypesWidget

### PR DESCRIPTION
**Pull request by andreas-unleash**
Implements ToggleTypeWidget

Refactored HealthWidget to it's own component

Signed-off-by: andreas-unleash <andreas@getunleash.ai>

## About the changes

Closes #

### Important files

Screenshot 2023-01-25 at 11 38 07

### CODE REVIEW

File: frontend/src/component/project/Project/ProjectInfo/HealthWidget.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import {
+    StyledArrowIcon,
+    StyledDivInfoContainer,
+    StyledDivPercentageContainer,
+    StyledLink,
+    StyledParagraphEmphasizedText,
+    StyledParagraphSubtitle,
+    StyledSpanLinkText,
+} from './ProjectInfo.styles';
+import PercentageCircle from '../../../common/PercentageCircle/PercentageCircle';
```
Comment (kwasniew) [inline frontend/src/component/project/Project/ProjectInfo/HealthWidget.tsx:10]: components/common without dots

File: frontend/src/component/project/Project/ProjectInfo/ToggleTypesWidget.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
+import { IFeatureToggleListItem } from '../../../../interfaces/featureToggle';
+import { useMemo } from 'react';
+import {
+    StyledCount,
+    StyledDivInfoContainer,
+    StyledParagraphGridRow,
+    StyledParagraphSubtitle,
+} from './ProjectInfo.styles';
+import { getFeatureTypeIcons } from '../../../../utils/getFeatureTypeIcons';
```
Comment (kwasniew) [inline frontend/src/component/project/Project/ProjectInfo/ToggleTypesWidget.tsx:9]: dots

File: frontend/src/component/project/Project/ProjectInfo/ToggleTypesWidget.tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+export interface IToggleTypesWidgetProps {
+    features: IFeatureToggleListItem[];
+}
+
+export const ToggleTypesWidget = ({ features }: IToggleTypesWidgetProps) => {
+    const { release, experiment, operational, kill, permission } =
+        useMemo(() => {
+            const release =
+                features?.filter(feature => feature.type === 'release')
```
Comment (kwasniew) [inline frontend/src/component/project/Project/ProjectInfo/ToggleTypesWidget.tsx:19]: can features be empty? the type doesn't indicate that

File: frontend/src/component/project/Project/ProjectInfo/ToggleTypesWidget.tsx [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+                    .length || 0;
+
+            return {
+                release,
+                experiment,
+                operational,
+                kill,
+                permission,
+            };
+        }, [features]);
```
Comment (kwasniew) [inline frontend/src/component/project/Project/ProjectInfo/ToggleTypesWidget.tsx:41]: this is a potential bug. react hooks should either depend on primitive types or use JSON.stringify in deps array

**Review by kwasniew — APPROVED** [review]
Still some warnings reported by github actions about unused items.

File: frontend/src/component/project/Project/ProjectInfo/ToggleTypesWidget.tsx (hunk 4 above)
Comment (andreas-unleash) [reply frontend/src/component/project/Project/ProjectInfo/ToggleTypesWidget.tsx:41]: Didn't know that, thanks

File: frontend/src/component/project/Project/ProjectInfo/ToggleTypesWidget.tsx (hunk 3 above)
Comment (andreas-unleash) [reply frontend/src/component/project/Project/ProjectInfo/ToggleTypesWidget.tsx:19]: It can, if it is a new project for example

Comment (kwasniew) [reply frontend/src/component/project/Project/ProjectInfo/ToggleTypesWidget.tsx:19]: but then it's empty array not undefined. So maybe the question mark is not needed.

**Pull request merged**